fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -532,13 +534,23 @@ public List<Expression> visitNamedExpressionSeq(NamedExpressionSeqContext namedC
      * @return List of OrderKey
      */
     @Override
-    public List<OrderKey> visitQueryOrganization(QueryOrganizationContext ctx) {
+    public QueryOrganization visitQueryOrganization(QueryOrganizationContext ctx) {
         return ParserUtils.withOrigin(ctx, () -> {
+            List<OrderKey> orderKeys;
             if (ctx.sortClause().ORDER() != null) {
-                return visit(ctx.sortClause().sortItem(), OrderKey.class);
+                orderKeys = visit(ctx.sortClause().sortItem(), OrderKey.class);
             } else {
-                return ImmutableList.of();
+                orderKeys = ImmutableList.of();
             }
+            long limit = Long.MAX_VALUE;
+            long offset = 0;
+            if (ctx.limitClause().LIMIT() != null) {
+                limit = Long.parseLong(ctx.limitClause().limit.getText());
+                if (ctx.limitClause().OFFSET() != null || ctx.limitClause().COMMA() != null) {
+                    offset = Long.parseLong(ctx.limitClause().offset.getText());
+                }
+            }
+            return new QueryOrganization(orderKeys, limit, offset);
```

Comment:
I think QueryOrganization is should be divided into LogicalSort and LogicalLimit.

Reply:
I also think so. We shouldn't add QueryOrganization, it's redundant.